lua/luasnip/init.lua

```diff
@@ -613,6 +662,9 @@ local function unlink_current_if_deleted()
 	end
 end
 
+---@alias ExitOutOfRegion fun(node: table): nil
```

Comment:
Yeah, I think we'll also need a `Node`-type, first for arguments like this, and second to communicate API (which is currently described [here](https://github.com/L3MON4D3/LuaSnip/blob/master/DOC.md#api))